src/components/AppCatalog/AppDetail/InstallAppModal.js

```diff
@@ -104,10 +104,13 @@ const InstallAppModal = (props) => {
       });
   }
 
-  const updateNamespace = (ns) => {
-    setNamespace(ns);
-    setNamespaceError(validateAppName(ns).message);
-  };
+  const updateNamespace = useCallback(
```

Comment:
This is the core of the fix. It makes sure `updateNamespace` does not get 'remade' every render, which causes the `useEffect` in the form to also fire again.

Reply:
ok so if I understand well `useCallback` you pass a function and an array of functions/variables from this context. If the argument of the function does not change the body of the function is not called, right?